lib/improver/tests/nowcasting/optical_flow/test_AdvectField.py

```diff
@@ -343,9 +343,10 @@ def test_unmasked_nans(self, warning_list=None):
         expected_data = np.full((4, 3), np.nan, dtype=np.float32)
         expected_data[2, 1:] = np.array([2., 3.])
         result = self.plugin.process(cube, self.timestep)
-        self.assertTrue(len(warning_list) == 1)
+        warning_msg = "contains unmasked NaNs"
```

Comment:
For the line below, where we are checking a particular position for a `UserWarning`, this should probably be replaced with the following, as is done in some of the other tests:
```python
        self.assertTrue(any(item.category == UserWarning
                            for item in warning_list))
```